src/module.rs

```diff
@@ -836,6 +836,8 @@ impl<'ctx> Module<'ctx> {
     /// assert_eq!(md_1[1].into_float_value(), f32_val);
     /// ```
     pub fn add_global_metadata(&self, key: &str, metadata: &MetadataValue<'ctx>) {
+        assert!(metadata.is_node());
```

Comment:
Should be a result rather than asserting

Reply:
Rather than being a result, can we make it an `Option`? It will make things a bit easier.

If we prefer a result, what the Error type should be?

Reply:
A Result makes more sense because it's reporting an error condition. But you can use the [ok](https://doc.rust-lang.org/std/result/enum.Result.html#method.ok) method to turn it into an option for your purposes and discard the error. Error type would be a 'static string for now. We'll eventually have a proper enum

Reply:
Fixed!